CHANGELOG.md

```diff
@@ -1,5 +1,26 @@
 # Changelog
 
+## 1.2.1 (2016-04-22)
+
+#### API
+
+Use engine-api to handle large number of API calls
+Update ContainerConfig to embed HostConfig and NetworkingConfig
+stop/restart/kill an non-existent container should returns 500 rather than 404
+Return an error when assertion fails in hijack
+Return error when Image Pull fails
+Fix image pull bug (wait until download finishes)
+
+#### Build
+
+Switch to go 1.6.2
```

Comment:
@vieux yeah, I know we had to revert docker/docker because of that, but not sure how common it is to connect to the Swarm API through a socket. Same as for Docker, or less common?